### DISCUSSION TITLE
Sample data

### DISCUSSION THREAD

**Discussion by roaldarbol** [talmolab/sleap#2009] [Help!]
Hi there, hope all's well!
I'm writing an R package for analysis of movement data ([animovement](https://github.com/roaldarbol/animovement)), and I'm about to write a reader for SLEAP CSV data. 
However, I don't have any exported sample data right at hand - would you happen to have some laying around? It would be great to have both single-2D, single-3D and multianimal data to allow import of them all. I mostly need to get snippets of all the headers and maybe the first 10 lines of data to run the tests I need. 
Thanks!

**Comment by talmo** [comment]
Hi @roaldarbol,

For sure -- [here's an example dataset](https://github.com/talmolab/sleap/tree/864c9941c1b092ac1d62c8461ac37d4d8a9b8fac/docs/notebooks/analysis_example) and [associated notebook](https://sleap.ai/notebooks/Analysis_examples.html) that walks through it.

The [`movement`](https://movement.neuroinformatics.dev/examples/load_and_explore_poses.html) folks also have example datasets you can check out.

Finally, I'd recommend checking out [`sleap-io`](https://io.sleap.ai/latest/) if you want to do any sort of I/O related things. This is a standalone library that has much less annoying dependencies than core SLEAP (e.g., DL and GUI deps).

---

Those datasets are all 2D/multi-animal in SLP or Analysis HDF5 formats, but you're welcome to poke around our [test fixtures](https://github.com/talmolab/sleap/tree/develop/tests/data) for other formats (e.g., CSV).

For 3D specifically: we haven't quite converged on a stable format for this yet since we're still actively developing it, but [sleap-anipose](https://github.com/talmolab/sleap-anipose) has the current backend for our aniposelib-based implementation of 3D. The tests folder there has example files and [keypoint-moseq has a loader for it](https://github.com/dattalab/keypoint-moseq/blob/9d8f66bdd24b490daef52d97f39f8e60f6028399/keypoint_moseq/io.py#L1173-L1192).


Let me know if there's anything we can help with!

Talmo

**Reply by roaldarbol** [reply]
Thanks for the reply @talmo!

I'm already using the `h5` files that movement uses; it was more for the natively exported csv files. Does that mean that https://github.com/talmolab/sleap/blob/develop/tests/data/csv_format/minimal_instance.000_centered_pair_low_quality.analysis.csv is a representative csv SLEAP exported file? What would look different if it was a multi-animal project?

It's a pure R package, so I can't rely on `sleap-io` for this (other than making my own test files 😉). I'm currently also implementing a reader for the `h5` files - I'll give you a heads-up when I have; then R folks should have a good place to go when wanting to works with your `h5` or `csv` files. 

I can't quite recall, are your `.slp` files simply stringent `h5` files? I.e. would it be possible to load them with a generic `h5` parser?

Good to know regarding 3D! I'll probably come back to them at a later point!

Thanks, and for now I think I'm all good - you're being super helpful answering questions, can't ask for more! 😃
Cheers,
Mikkel

**Reply by roaldarbol** [reply]
Didn't take as long as expected - now the `h5` reader is implemented! https://github.com/roaldarbol/animovement/pull/50
Just got the `csv` reader to go then!
